ben-basten/immich
# feat(web,a11y): consolidate BaseModal into FullScreenModal

## Description

General changes:

- All modals have been migrated to use the `FullScreenModal` component!
- Added a "sticky-bottom" slot to the FullScreenModal, reducing the boilerplate/duplicated code to add action buttons
- New subtle horizontal rule between the main modal content and action buttons, to make it easier to see the sticky bottom section
- Limit the height of the modal to 900px or 95vh, whichever is smaller
- Action buttons: 
     - standardized to always span full width
     - stack in narrow views, for easier use and preventing horizontal scrolling
     - stick to the bottom of the modal - they are always visible, even if there's vertical scrolling

Accessibility improvements:

- Cancel button in confirmation modals is consistently the "secondary" color
- Scroll padding prevents focus from hiding behind the sticky bottom

## Screenshots

<details><summary>Add to album / Add to shared album</summary>
<p>

**Mobile**

![Screenshot from 2024-04-12 19-15-50](https://github.com/ben-basten/immich/assets/45583362/0b9fb24b-eaf4-42ce-b539-582b7f826877)

**Desktop - hovering over option**

Now that the content of this modal doesn't span the modal edge-to-edge, I added an xl radius to the buttons to make them look nicer on hover.

![Screenshot from 2024-04-12 19-35-48](https://github.com/ben-basten/immich/assets/45583362/2b78cd6f-1125-4cb6-9a4d-ddd342bfa032)

</p>
</details> 

<details><summary>Confirmation Dialogue</summary>
<p>

**Mobile**

![Screenshot from 2024-04-12 19-12-41](https://github.com/ben-basten/immich/assets/45583362/47912d21-33a0-4ccc-91f4-8e6a4bfd21bf)

**Desktop**

![Screenshot from 2024-04-12 19-10-39](https://github.com/ben-basten/immich/assets/45583362/171d5849-fb1c-4ea5-9736-29a0a39fc5c1)

</p>
</details> 

<details><summary>Edit date and time</summary>
<p>

**Desktop - dark**

![desktop-dark](https://github.com/ben-basten/immich/assets/45583362/9c288059-08c4-488a-b2ba-0fb6775bb3f3)

**Desktop - light**

![desktop-light](https://github.com/ben-basten/immich/assets/45583362/cdc6826c-375e-452e-b7f9-6e38a0e57b6f)

**Mobile - dark**

![mobile-dark](https://github.com/ben-basten/immich/assets/45583362/75d6eed3-17fb-4775-9eec-8bb8e59931c6)

**Mobile - light**

![mobile-light](https://github.com/ben-basten/immich/assets/45583362/fb8b7961-f191-433d-8244-87a29fb073b8)

</p>
</details> 

<details><summary>External Libraries</summary>
<p>

**Desktop**

![desktop](https://github.com/ben-basten/immich/assets/45583362/4d6b4832-0442-4b71-ba7d-3ca54ebc1fae)

**Mobile**

![mobile](https://github.com/ben-basten/immich/assets/45583362/8da84b6a-a9f7-4b7f-9ec8-3c65e96bdf43)

</p>
</details> 

<details><summary>Sharing options</summary>
<p>

**Desktop - hovering over option**


Now that the content of this modal doesn't span the modal edge-to-edge, I added an xl radius to the buttons to make them look nicer on hover.

![desktop-hover](https://github.com/ben-basten/immich/assets/45583362/eebb4505-6ecc-4532-8f9f-a9e212c22763)

**Mobile**

![mobile](https://github.com/ben-basten/immich/assets/45583362/1feb9b8f-e185-4188-9ad5-f6308b04b605)

</p>
</details> 

<details><summary>Create user</summary>
<p>

**Desktop**

![desktop](https://github.com/ben-basten/immich/assets/45583362/051e3d44-647c-4e65-8f2f-b2f178ae37c5)

**Mobile**

![mobile](https://github.com/ben-basten/immich/assets/45583362/aa4cb856-cffd-414f-a21d-fc48770748c4)

</p>
</details> 

<details><summary>Edit user</summary>
<p>

**Desktop**

![desktop](https://github.com/ben-basten/immich/assets/45583362/aca122a1-4b47-4326-9be3-92b25c405aa3)

**Mobile**

![mobile](https://github.com/ben-basten/immich/assets/45583362/eb84304f-6e6f-4be1-9914-434231a86645)

</p>
</details> 

<details><summary>Create shared link</summary>
<p>

**Desktop**

![desktop](https://github.com/ben-basten/immich/assets/45583362/27aaa35d-38d7-41e7-9565-830a17cecf49)

**Mobile**

![mobile](https://github.com/ben-basten/immich/assets/45583362/bcd0c849-5dda-4925-9f0f-538be8754ba9)

</p>
</details> 

<details><summary>Add partner</summary>
<p>

**Desktop**

![Screenshot from 2024-04-12 19-29-32](https://github.com/ben-basten/immich/assets/45583362/1eb761e9-604a-489d-a27f-fbd1edb04bf0)

**Mobile**

![Screenshot from 2024-04-12 19-17-19](https://github.com/ben-basten/immich/assets/45583362/c410841b-c0a9-41e1-8074-58aa68f632fe)

</p>
</details> 

## How Has This Been Tested?



Open the following affected modals to verify they work as expected:

- Edit album
- Album sharing "Options"
- Invite to album
- Add to album / Add to shared album
- Merge people
- Set date of birth
- API key creation / editing / copy secret
- Create user form
- Edit user form
- Add exclusion pattern
- Add import path
- Select library owner
- Map settings
- Create shared link
- Set profile picture
- Version announcement
- Slideshow settings
- Add partner (partner selection)
- Change name

## Checklist:

- [x] npm run lint (linting via ESLint)
- [x] npm run format (formatting via Prettier)
- [x] npm run check:svelte (Type checking via SvelteKit)
- [x] npm test (unit tests)
